lib/spack/spack/main.py

```diff
@@ -976,6 +967,15 @@ def _main(argv=None):
     # ensure options on spack command come before everything
     setup_main_options(args)
 
+    # -h, -H, and -V are special as they do not require a command, but
+    # all the other options do nothing without a command.
+    if args.version:
+        print(get_version())
+        return 0
+    elif args.help:
+        sys.stdout.write(parser.format_help(level=args.help))
+        return 0
+
```

Comment:
I think this works because you're relying on the config setup in `setup_main_options`, right?  Are you using `spack -C <config> ...` or something, where your extension setup is in its own config directory?  That only gets set up in `setup_main_options`, so `spack.extensions.get_command_path()` won't see it until after `setup_main_options`.

Should we move this even further down though?  Environment setup happens just below this line, and without that (for the same reason) you won't be able to put extension config in an environment and have it appear in `spack -h` output.